features/auth/components/LoginForm.tsx

```diff
@@ -0,0 +1,105 @@
+'use client'
+
+import React, { useState, useTransition } from 'react'
+import { CardWrapper } from './cardWrapper'
+import {
+	Form,
+	FormControl,
+	FormField,
+	FormItem,
+	FormLabel,
+	FormMessage,
+} from '@/components/ui/form'
+
+import { useForm } from 'react-hook-form'
+
+import { Input } from '@/components/ui/input'
+import { z } from 'zod'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { LoginSchema } from '../schemas/LoginSchema'
+import { Button } from '@/components/ui/button'
+import { ErrorMessage, SuccessMessage } from './ui'
+import { login } from '@/actions/login'
+
+type LoginFormData = z.infer<typeof LoginSchema>
+
+export const LoginForm = () => {
+	const [success, setSuccess] = useState('')
+	const [error, setError] = useState('')
+	const [isPending, startTransition] = useTransition()
+
+	const form = useForm<LoginFormData>({
+		resolver: zodResolver(LoginSchema),
+		defaultValues: { email: '', password: '' },
+	})
+
+	const onSubmit = (values: LoginFormData) => {
+		setSuccess('')
+		setError('')
+		startTransition(() => {
+			login(values)
+				.then(data => {
+					if (data?.error) {
+						form.reset()
+						setError(data.error)
+					}
+
+					if (data?.success) {
```

Comment:
Obsłużyc cały błąd w catch (Jak będziesz throwował w action to w .catch(e) masz e.message )